src/main/java/org/spongepowered/common/event/SpongeCommonEventFactory.java

```diff
@@ -218,27 +218,19 @@ public static CollideEntityEvent callCollideEntityEvent(net.minecraft.world.Worl
     }
 
     public static boolean handleChangeBlockEventPre(IMixinWorldServer worldIn, BlockPos pos) {
-        return handleChangeBlockEventPre(worldIn, pos, null);
+        return handleChangeBlockEventPre(worldIn, pos, ImmutableList.of(new Location<>((World) worldIn, pos.getX(), pos.getY(), pos.getZ())));
     }
 
-    public static boolean handleChangeBlockEventPre(IMixinWorldServer worldIn, BlockPos pos, BlockSnapshot source) {
+    public static boolean handleChangeBlockEventPre(IMixinWorldServer worldIn, BlockPos pos, List<Location<World>> list) {
         final CauseTracker causeTracker = worldIn.getCauseTracker();
-        PhaseData data = causeTracker.getCurrentPhaseData();
-        if (source == null) {
-            source = data.context.getSource(BlockSnapshot.class).orElse(null);
-            if (source == null) {
-                // safety measure
-                return false;
-            }
-        }
+        BlockSnapshot source = ((World) worldIn).createSnapshot(pos.getX(), pos.getY(), pos.getZ());
```

Comment:
The source is intended to be captured beforehand, if in any case it was captured beforehand. Creating snapshots is potentially expensive as is (for tile entities). If at all possible, retaining the previous logic is preferred. Unless @bloodmc has other opinions.